lib/modules/manager/circleci/schema.ts

```diff
@@ -9,9 +9,17 @@ export const CircleCiJob = z.object({
   docker: z.array(CircleCiDocker).optional(),
 });
 
+export type CircleCiOrb = z.infer<typeof CircleCiOrb>;
+export const CircleCiOrb = z.object({
+  executors: z.record(z.string(), CircleCiJob).optional(),
+  jobs: z.record(z.string(), CircleCiJob).optional(),
+  orbs: z.record(z.unknown()).optional(),
```

Comment:
I don't have a great handle on zod, so if there's a better way to do this recursive type check please let me know!

Reply:
Can you give an example?. I will try to form a better schema from it.

Reply:
The `config5.yml` fixture partially shows it, but this is another example of the recursiveness:
```
version: 2.1

orbs:
  nodejs: circleci/node@5.2.0

  myorb:
    jobs:
      test_image:
        docker:
          - image: cimg/python:3.7
        steps:
          - checkout

    orbs:
      nodejs: circleci/node@5.2.0

      myotherorb:
        orbs:
          nodejs: circleci/node@5.2.0

        executors:
          python:
            docker:
              - image: cimg/python:3.9

workflows:
  Test:
    jobs:
      - myorb/test_image

```

Reply:
```ts
type CircleCIOrbType = z.ZodObject<{
  jobs?: z.ZodRecord<z.ZodString, CircleCIJob>;
  orbs?: z.ZodRecord<z.ZodString, z.ZodString | z.ZodLazy<z.ZodType<CircleCIOrbType>>>;
  executors?: z.ZodRecord<z.ZodString, CircleCIJob>;
}>;

const CircleCIOrb : z.ZodType<CircleCIOrbType > = z.lazy(() => 
  z.object({
    jobs: z.record(CircleCIJob).optional(),
    orbs: z.record(z.union([z.string(), CircleCIOrb])).optional(),
    executors: z.record(CircleCIJob).optional(),
  })
);
```

Reply:
thanks! I was able to get it working with a bit of help from https://github.com/colinhacks/zod/issues/1500#issuecomment-2324304993. let me know what you think of d8f2826387036b32950361c6524b89b6a940acd3.